sympy/tensor/arraypy.py

```diff
@@ -0,0 +1,1414 @@
+
+        Examples
+        ========
+
+        """
+        # forming list of tuples for Arraypy constructor of type
+        # a = Arraypy( [(a, b), (c, d), ... , (y, z)] )
+        arg = [(self.start_index[i], self.end_index[i])
+               for i in range(self._rank)]
+        res = Arraypy(arg)
+
+        idx = self.start_index
+        for i in range(len(self)):
+            res[idx] = self[idx] / other
+            idx = self.next_index(idx)
+
+        return res
+
+    # imethods. += -= *= /=
+    def __iadd__(self, other):
+        return self + other
+
+    def __isub__(self, other):
+        return self - other
+
+    def __imul__(self, other):
+        return self * other
+
+    def __itruediv__(self, other):
+        return self / other
+
+    __rmul__ = __mul__
+    __div__ = __truediv__
+
+    def __eq__(self, other):
+        """
+        Overloads '=='.
+        Arraypy instances can be compared to each other.
+        Instances equal if they have same shape, indexes and data.
+
+        Examples
+        ========
+
+        >>> from sympy import Arraypy
+        >>> a = Arraypy((2, 3))
+        >>> b = Arraypy((2, 3))
+        >>> a == b
+        True
+        >>> c = a.reshape((3,2))
+        >>> c == b
+        False
+        >>> a[0,0] = 1
+        >>> b[0,0] = 2
+        >>> a == b
+        False
+        """
+        if not isinstance(other, Arraypy):
+            return False
+            #raise TypeError('Compared instances must be Arraypy type')
+        if (self.shape != other.shape or self.start_index !=
+                other.start_index or self.end_index != other.end_index):
+            return False
+        idx = self.start_index
+        for i in range(len(self)):
+            if (self[idx] != other[idx]):
+                return False
+            idx = self.next_index(idx)
+
+        return True
+
+    def __getitem__(self, index):
+        """Allows to get items from arraypy.
+
+        Examples
+        ========
+
+        >>> from sympy.tensor.arraypy import Arraypy, list2arraypy
+        >>> a = list2arraypy(list(range(4)), (2,2))
+        >>> print (a)
+        0 1
+        2 3
+        >>> print(a[0,0])
+        0
+        >>> print(a.__getitem__((1,1)))
+        3
+
+        """
+        if isinstance(index, int):
+            index = (index,)
+        if len(index) != self._rank:
+            raise ValueError('Wrong number of array axes')
+
+        # check if input index can exist in current indexing
+        for i in range(self._rank):
+            if index[i] >= self._end_index[
+                    i] or index[i] < self._start_index[i]:
+                raise ValueError('Value ' + str(i) + ' out of border')
+
+        # returning element. If array is sparse and index not in dictionary
+        # then return '0'
+        try:
+            if self._sparse:
+                if index in self._output:
+                    return self._output[index]
+                else:
+                    return 0
```

Comment:
Suppose you have an Arraypy _a_, _a_ is sparse and _a[3]_ is zero.

`(1+a[3])/2` should be `1/2` (SymPy object), not `0.5` as with Python numbers.